subsystem/disconnected_cluster_test.go

```diff
@@ -0,0 +1,343 @@
+					Name:             swag.String("disconnected-infraenv-on-ready"),
+					OpenshiftVersion: openshiftVersion,
+					PullSecret:       swag.String(pullSecret),
+					SSHAuthorizedKey: swag.String(utils_test.SshPublicKey),
+					ImageType:        models.ImageTypeDisconnectedIso,
+					ClusterID:        &clusterID,
+				},
+			})
+			Expect(err).NotTo(HaveOccurred())
+			infraEnv = infraEnvResp.GetPayload()
+
+			By("Verifying cluster remains in ready state (not changed to disconnected)")
+			Consistently(func() string {
+				clusterResp, err := utils_test.TestContext.UserBMClient.Installer.V2GetCluster(ctx, &installer.V2GetClusterParams{
+					ClusterID: clusterID,
+				})
+				Expect(err).NotTo(HaveOccurred())
+				return swag.StringValue(clusterResp.GetPayload().Status)
+			}, 5*time.Second, 1*time.Second).Should(Equal(models.ClusterStatusReady))
+		})
+	})
+
+	Context("OVE discovery.ign download", func() {
+		It("download discovery.ign for disconnected-iso infraenv bound to cluster", func() {
+			By("Creating a cluster with 0 hosts")
+			cluster, err := utils_test.TestContext.UserBMClient.Installer.V2RegisterCluster(ctx, &installer.V2RegisterClusterParams{
+				NewClusterParams: &models.ClusterCreateParams{
+					Name:              swag.String("ove-discovery-cluster"),
+					OpenshiftVersion:  swag.String(openshiftVersion),
+					PullSecret:        swag.String(pullSecret),
+					BaseDNSDomain:     "example.com",
+					ControlPlaneCount: swag.Int64(0),
+				},
+			})
+			Expect(err).NotTo(HaveOccurred())
+			clusterID = *cluster.GetPayload().ID
+
+			By("Registering a disconnected-iso InfraEnv")
+			infraEnvResp, err := utils_test.TestContext.UserBMClient.Installer.RegisterInfraEnv(ctx, &installer.RegisterInfraEnvParams{
+				InfraenvCreateParams: &models.InfraEnvCreateParams{
+					Name:             swag.String("ove-discovery-infraenv"),
+					OpenshiftVersion: openshiftVersion,
+					PullSecret:       swag.String(pullSecret),
+					SSHAuthorizedKey: swag.String(utils_test.SshPublicKey),
+					ImageType:        models.ImageTypeDisconnectedIso,
+					ClusterID:        &clusterID,
+				},
+			})
+			Expect(err).NotTo(HaveOccurred())
+			infraEnv = infraEnvResp.GetPayload()
+			infraEnvID = *infraEnv.ID
+
+			By("Downloading discovery.ign")
+			file, err := os.CreateTemp("", "tmp")
+			Expect(err).NotTo(HaveOccurred())
+			defer os.Remove(file.Name())
+
+			_, err = utils_test.TestContext.UserBMClient.Installer.V2DownloadInfraEnvFiles(ctx,
+				&installer.V2DownloadInfraEnvFilesParams{
+					InfraEnvID: infraEnvID,
+					FileName:   "discovery.ign",
+				}, file)
+
+			// The download may fail if openshift-install is not available in the test environment
+			// This is expected and consistent with how unit tests handle this scenario
+			if err == nil {
+				By("Verifying file is not empty")
+				s, err := file.Stat()
+				Expect(err).NotTo(HaveOccurred())
+				Expect(s.Size()).ShouldNot(Equal(0))
```

Comment:
Can we also do some basic sanity tests on the content? As in: https://github.com/openshift/assisted-service/blob/master/subsystem/infra_env_test.go#L355
E.g. check ssh-key/pull-secret are included, etc.